cucumber-core/src/test/java/io/cucumber/core/eventbus/IncrementingUuidGeneratorTest.java

```diff
@@ -0,0 +1,62 @@
+package io.cucumber.core.eventbus;
+
+import io.cucumber.core.exception.CucumberException;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Field;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class IncrementingUuidGeneratorTest {
+    @Test
+    void generates_different_non_null_uuids() {
+        // Given
+        UuidGenerator generator = new IncrementingUuidGenerator();
+        UUID uuid1 = generator.get();
+
+        // When
+        UUID uuid2 = generator.get();
+
+        // Then
+        assertNotNull(uuid1);
+        assertNotNull(uuid2);
+        assertNotEquals(uuid1, uuid2);
+    }
+
+    @Test
+    void raises_exception_when_out_of_range() throws NoSuchFieldException, IllegalAccessException {
+        // Given
+        UuidGenerator generator = new IncrementingUuidGenerator();
+        Field counterField = IncrementingUuidGenerator.class.getDeclaredField("counter");
+        counterField.setAccessible(true);
+        AtomicLong counter = (AtomicLong) counterField.get(generator);
+        counter.set(Long.MAX_VALUE - 1);
```

Comment:
Consider a package private constructor to initialize a certificain problem for testing instead of using reflection.

Reply:
Used package private visibility.